docs/Training-Generalization-Learning.md

```diff
@@ -0,0 +1,105 @@
+# Training Generalized Reinforcement Learning agents
+
+Reinforcement learning has a rather unique setup as opposed to supervised and
+unsupervised learning. Agents here are trained and tested on the same exact environment,
+which is analogous to a model being trained and tested on an identical dataset in supervised learning! This setting results in overfitting; the inability of the agent to generalize to slight tweaks or variations in the environment. This is problematic in instances when environments are randomly instantiated with varying properties. To make agents more robust, we train an agent over multiple variations of the environment. The agent is trained with the intent that it learns to maintain a minimum performance regardless of the environment variant and that it generalizes to maintain this in unseen future variants of the environment.
+
+Ball scale of 0.5          |  Ball scale of 4
+:-------------------------:|:-------------------------:
+![](images/3dball_small.png)  |  ![](images/3dball_big.png)
+
+_Variations of the 3D Ball environment._
+
+To vary the environments, we first decided what the reset parameters for the environment. These parameters are known as `Reset Parameters`, introduced in [Curriculum Learning](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Training-Curriculum-Learning.md). In the 3D ball environment example displayed in the figure above, the reset parameters are `gravity`, `ball_mass` and `ball_scale`.
+
+## In-practice
+
+To test the effectiveness of this training procedure, we train 3 models over 50000 steps:
+    1. Model Trained on Default reset parameter values (Default model).
```

Comment:
This numbering does not look good in the markdown. I think you need to skip a line somewhere for it to be a numbered list.